src/zenml/integrations/dash/visualizers/pipeline_run_lineage_visualizer.py

```diff
@@ -0,0 +1,344 @@
+                                            style={
+                                                "width": "100%",
+                                                "height": "800px",
+                                            },
+                                            zoom=1,
+                                        )
+                                    ]
+                                ),
+                                dbc.Row(
+                                    [
+                                        dbc.Button(
+                                            "Reset",
+                                            id="bt-reset",
+                                            color="primary",
+                                            className="me-1",
+                                        )
+                                    ]
+                                ),
+                            ]
+                        ),
+                        dbc.Col(
+                            [
+                                dcc.Markdown(id="markdown-selected-node-data"),
+                            ]
+                        ),
+                    ]
+                ),
+            ],
+            className="p-5",
+        )
+
+        @app.callback(
+            Output("markdown-selected-node-data", "children"),
+            Input("cytoscape", "selectedNodeData"),
+        )
+        def display_data(data_list) -> str:
+            """Callback for the text area below the graph"""
+            if data_list is None:
+                return "Click on a node in the diagram."
+
+            text = ""
+            for data in data_list:
+                text += f'## {data["execution_id"]} / {data["name"]}' + "\n\n"
+                if data["type"] == "artifact":
+                    for item in [
+                        "artifact_data_type",
+                        "is_cached",
+                        "producer_step_id",
+                        "parent_step_id",
+                        "uri",
+                    ]:
+                        text += f"**{item}**: {data[item]}" + "\n\n"
+                elif data["type"] == "step":
+                    text += "### Inputs:" + "\n\n"
+                    for k, v in data["inputs"].items():
+                        text += f"**{k}**: {v}" + "\n\n"
+                    text += "### Outputs:" + "\n\n"
+                    for k, v in data["outputs"].items():
+                        text += f"**{k}**: {v}" + "\n\n"
+                    text += "### Params:"
+                    for k, v in data["parameters"].items():
+                        text += f"**{k}**: {v}" + "\n\n"
+            return text
+
+        @app.callback(
+            [Output("cytoscape", "zoom"), Output("cytoscape", "elements")],
+            [Input("bt-reset", "n_clicks")],
+        )
+        def reset_layout(n_clicks) -> List[int, List[Dict, Any]]:
```

Comment:
Not sure what the return type should be here, but there are two issues with it:
- a list only takes a single type as its generic type (if it's a list of ints or lists, we should use `List[Union[int, List[...]]]`
- `Dict` needs to have annotations for key/value types, e.g. `Dict[str, Any]` or similar

Reply:
It seems like this could work:

```suggestion
        def reset_layout(n_clicks: str) -> List[Union[int, List[Dict[str, Dict[str, str]]]]]::
```

Question is, how deep do we want to go when typing even more nested data?